src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
@parjong a little correction to the scenario you have described (managed method A calls a native method B, and B throws a managed exception that A will handle). There doesn't have to be the transition frame. If the native method is a FCall like e.g. IL_Throw, there is the HelperMethodFrame instead. 

I believe the fix to the issue with the result from VirtualUnwindToFirstManagedCallFrame not matching the other one should be to make everything consistent. That means, if we have a context of a managed frame, the ESP should always have the same semantics. No matter whether we got it from the native unwinder that has reached the first managed frame or if we got it from managed unwinder that has unwound from one managed frame to another one.
As we have already discussed here, there are two options. Either modify the OOPStackUnwinderX86::VirtualUnwind so that it returns context with ESP being the ESP at the call instruction or augment the result of the PAL_VirtualUnwind when it reaches a managed frame so that the ESP is the ESP before the stack arguments were pushed.

Having context for managed frames with the ESP semantic sometimes one way and sometimes the other is error prone and hard to reason about.